src/main/java/com/autotune/common/datasource/DataSourceMetadataOperator.java

```diff
@@ -51,19 +51,20 @@ public void createDataSourceMetadata(DataSourceInfo dataSourceInfo) {
          * TODO - Process cluster metadata using a custom query
          */
         try {
-            JsonArray namespacesDataResultArray =  op.getResultArrayForQuery(dataSourceInfo.getUrl().toString(), PromQLDataSourceQueries.NAMESPACE_QUERY);
+            String datasourceName = dataSourceInfo.getName();
```

Comment:
It looks like there is code duplication between create and update, can we move the common code to a function

Reply:
Have moved the common code between the create and update methods i.e processing workload and container  data to `updateWorkloadAndContainerMetadataInfo()`.
Namespace data processing approach is different hence retaining the functionality in the respective functions.